server/consumer.go

```diff
@@ -1523,13 +1515,43 @@ func (o *consumer) updateConfig(cfg *ConsumerConfig) error {
 		o.maxdc = uint64(cfg.MaxDeliver)
 	}
 
+	// Set InactiveThreshold if changed.
+	if cfg.InactiveThreshold != o.cfg.InactiveThreshold {
+		cfg.InactiveThreshold = o.updateInactiveThresholdLocked(cfg.InactiveThreshold, true)
+	}
+
 	// Record new config for others that do not need special handling.
 	// Allowed but considered no-op, [Description, SampleFrequency, MaxWaiting, HeadersOnly]
 	o.cfg = *cfg
 
 	return nil
 }
 
+// updateInactiveThresholdLocked sets the dthres and either resets the
+// dtmr (if there is one) or forcibly starts one, depending on the value
+// of the "forceStart" parameter. It then returns what the final value
+// for InactiveThreshold will be (so that the jittery default for ephemeral
+// consumers can be captured in config).
+func (o *consumer) updateInactiveThresholdLocked(threshold time.Duration, forceStart bool) time.Duration {
+	started := o.dtmr != nil
+	defer func() {
+		if forceStart || started {
+			stopAndClearTimer(&o.dtmr)
+			o.dtmr = time.AfterFunc(o.dthresh, func() { o.deleteNotActive() })
+		}
+	}()
+	if threshold > 0 {
+		o.dthresh = threshold
+	} else if !o.isDurable() {
+		// Ephemerals will always have inactive thresholds.
+		// Add in 1 sec of jitter above and beyond the default of 5s.
+		o.dthresh = JsDeleteWaitTimeDefault + time.Duration(rand.Int63n(1000))*time.Millisecond
+		// Only stamp config with default sans jitter.
+		return JsDeleteWaitTimeDefault
+	}
```

Comment:
So what if the user uses a durable consumer and had a threshold and now wants to set it to 0. That would never been reset to 0. If we support that case, then we would need to set it here and also change the defer func() to not reset the o.dtmr if threshold is 0.

Reply:
Also, as it stands, any update to the consumer reset its inactivity timer, which may be ok, but that will delay the cleanup (even if there is no interest or no pull made for pull consumers). I think it is "ok" because if the consumer is updated it is likely that one wants the consumer to stick around, but let's make sure we understand the consequences.

Reply:
So I think setting it to `0` is an easy enough fix and it would seem like it's reasonable to reset the timer and not restart it if that's the case.

Modifying the `defer` is more complicated. I did have different behaviour there originally but most permutations seem to break the `TestJetStreamSuperClusterEphemeralCleanup` test. I'm not 100% sure I understand yet what that test does or why we would expect a timer to not be running there.

Reply:
It broke because we used to change the threshold value but with existing code the timer was known to not be set yet. With your changes (and calling that new function), it would be started as right there (in previous commit you were calling with forceStart==false, so were not starting it).

It could be that the test needs updating if the timer *should* have been started. Sometimes - unfortunately - tests are written with the assumption that the current code behaves as it should :-)

Reply:
I think actually I might have figured out a solution. I've left the setting durable to `0` in but it appears to work and pass either way with/without that, so easy enough to revert if there's no appetite to support that.

Reply:
Bah. Spoke too soon.